### ISSUE TITLE
Release alpha version for sharing with MIT

### ISSUE THREAD

**Issue by jwodder** [dandi/s3invsync#45]
Prerequisites:

- [x] #4
- [x] #5
- [x] #6

**Comment by jwodder** [comment]
@yarikoptic I've just released v0.1.0-alpha.  Be sure to pass the `--release` option to `cargo build` and `cargo run` in order to enable optimizations.

**Comment by yarikoptic** [comment]
@satra  -- it seems indeed in the past those inventory files were not available annonymously:

```shell
(3.13) (py3.13) yoh@typhon:/mnt/DATA/data/tmp/yoh/dandi$ ./s3invsync --date 2021-01-01T00-00Z s3://dandiarchive/dandiarchive/dandiarchive/ dump
2024-12-02T21:17:39.534474929-05:00  INFO s3invsync: Determining region for S3 bucket ... bucket=dandiarchive
2024-12-02T21:17:39.612263844-05:00  INFO s3invsync: Found S3 bucket region bucket=dandiarchive region=us-east-2
2024-12-02T21:17:39.612638302-05:00  INFO s3invsync: Fetching manifest ...
2024-12-02T21:17:39.612655669-05:00  INFO s3invsync::s3: Getting manifest for timestamp timestamp=2021-01-01T00-00Z
2024-12-02T21:17:39.612674807-05:00 DEBUG get_manifest{when=2021-01-01T00-00Z}: s3invsync::s3: Fetching manifest.checksum file
Error: failed to get object at s3://dandiarchive/dandiarchive/dandiarchive/2021-01-01T00-00Z/manifest.checksum

Caused by:
    0: service error
    1: unhandled error (AccessDenied)
    2: Error { code: "AccessDenied", message: "Access Denied", s3_extended_request_id: "EsaX9csxfPO6Lr+hqNVW5cPVE3sCDW2l5OFjj0UmUK1+eBKDlebKGoiMUHzwnePGhAgIone7LfAldmNh1ZctjPlN/uPK1fosRHEap0KlPPQ=", aws_request_id: "BGD5Y20876A9WADA" }
```
but then 

@kabilar et al who might decide to try s3invsync (can get a statically built version from http://www.oneukrainian.com/tmp/s3invsync . sha256: 614ce8586079e20ecedf938c0543fd14c02b379e893f1fb811b86b4444407e8e) , a sample invocation to "test speeds" and either "local IO would not become a bottleneck with fast network", try running

```
/s3invsync --date 2022-01-01T00-00Z s3://dandiarchive/dandiarchive/dandiarchive/ dump
```

at that date we didn't yet have explosion of zarrs (came some time after):

```
dandi@drogon:/mnt/backup/dandi/dandiarchive-inventory$ du -sck dump/202*-01-01*
8       dump/2020-01-01T00-00Z
8       dump/2021-01-01T00-00Z
8       dump/2022-01-01T00-00Z
72      dump/2023-01-01T01-00Z
76      dump/2024-01-01T01-00Z
```

so you should be able to fill up the pipe quite nicely. E.g. I was observing download speeds 300-600MBps but then it slowed down and after I interrupted after few minutes, write IO was still busy at 300 MBps (I guess that is max write speed I could afford) .

A sample invocation on some sample zarrs (but might take awhile to locate them in inventory first) is

```
duct ./s3invsync --path-filter '^dandiarchive/zarr/ff'  s3://dandiarchive/dandiarchive/dandiarchive/ dump
```

which I am running now with `duct` of ours (attn @asmacdo ) to see what we get in terms of CPU etc through time.
